src/tests/testHtmlQuote.cc

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/cppunit.h"
+#include "html_quote.h"
+#include "unitTestMain.h"
+
+#include <cstring>
+#include <iostream>
+
+class testHtmlQuote: public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE(testHtmlQuote);
+    CPPUNIT_TEST(test_html_quote_cstr);
+    // CPPUNIT_TEST(testPerformance);
```

Comment:
I actually plan to remove it before merging. I mostly used it to quantify the speed improvements coming out of using CharacterSet, but that's not really relevant for correctness, and to show reviewers what I used so that they can pick nits at it.

Reply:
Thank you for showing what you use for performance testing. Please keep it until we are ready to merge this PR (and completely remove it then). I would also enable it until then, but I did not study the details and do not insist on it.

Reply:
There is little point in leaving it enabled, as it's tricky to obtain a baseline measure.
It might come in handy to assess the efficiency of an alternate
quoting implementation that uses SBufs directly

Reply:
> There is little point in leaving it enabled

Just to clarify: The point of leaving it enabled is to allow others working on this PR to run this code without modifying sources. This added convenience is minor, of course.

Reply:
After all performance related improvements are completed (and PR description updated accordingly), please do not forget to remove testPerformance().

Reply:
Done